04_Memory_Management/01_Overview.md

```diff
@@ -45,26 +64,39 @@ Each of the layers has a dedicated chapter, however we'll start with a high leve
 - If using a higher half kernel, the upper half of every VMM will be identical, and contain the protected kernel code and data.
 - Can present other resources via virtual memory to simplify their interface, like memory mapping a file or inter-process communication.
 
+Similarly to paging there are some things we need to consider depending on our future decisions:
+
+* If we are going to support multiple address spaces, we need to initialize a different VMM for every task, so all the initialization/allocation/free function should be aware of which is the VMM that needs to be updated.
+* In case we want to implement User and Supervisor support, a good idea is to have separate address space for the user processes/threads and the supervisor one. Usually the supervisor address space is in the higher half and , as in it starts at the lowest address of the higher half of the address space (in x86_64 and ricsv it is starting from: 0xFFFF800000000000)
```

Comment:
I think some words are missing here 😆

Reply:
the problem now is what words? lol i don't remember what i wanted to write!

Reply:
`Usually the supervisor address space is in the higher half and , as in it starts at the`

Reply:
I think you were going to say something like `and the user address space lives in the lower half`

Reply:
ah yeah i think it was it XD